pocs/utils/rs232.py

```diff
@@ -109,16 +115,20 @@ def write(self, value):
 
         return response
 
-    def read(self):
-        """
-        Reads value using readline
-        If no response is given, delay and then try to read again. Fail after 10 attempts
+    def read(self, retry_limit=None, retry_delay=None):
+        """Reads next line of input using readline.
+
+        If no response is given, delay for retry_delay and then try to read
+        again. Fail after retry_limit attempts.
         """
+        assert not self._start_needed
```

Comment:
I don't see where this comes from or get what it is doing.

Reply:
Removed.